### PR TITLE
Login/Registration Encryption

### LINKED ISSUE #11: Login/Registration Encryption
The registration API needs to encrypt the password data before inserting it into the database. The login API needs to then use encryption to validate the password.

### PR DESCRIPTION
**Pull request by ishan0102**
### Problem
Raw passwords were being stored in the database.

### Solution
Encrypt the passwords using `Flask-Bcrypt`. Passwords are encrypted upon registration by generating a hash and salt through Bcrypt and the hash is stored in the database. On login, the hashed password is compared against a new hash, generated from the password the user inputted. If these hashes match, the user is logged in.

### Testing
I tested this by sending registration and login requests through Postman to make sure passwords could be decrypted correctly during login. I also checked the mongo database to see if the passwords were stored in encrypted form correctly.

### Notes
`Flask-Bcrypt` requires hashes to be decoded to UTF-8 before storing them into the database. This is important if we want to encrypt anything else. I've left a code snippet below showing how to do it.

```py
pw_hash = bcrypt.generate_password_hash(‘michael2’).decode(‘utf-8’)
bcrypt.check_password_hash(pw_hash, 'michael2') # returns True
```

Closes #11

### CODE REVIEW

File: server/api/routes/users.py [hunk 1]
```diff
@@ -30,7 +31,7 @@ def login():
             'email': 'No account with this email exists.',
             'password': ''
         }, 404
-    if user.password != req['password']:
+    if not bcrypt.check_password_hash(user.password, req['password']):
         return {
             'email': 'An account was found but the password is incorrect.',
             'password': ''
```
Comment (JimothyGreene) [inline server/api/routes/users.py]: This isn't relevant to this PR, but I didn't notice earlier. This should probably be a password error saying that the password is incorrect

**Review by JimothyGreene — APPROVED** [review]
Seems good. Just make sure the commit I pushed doesn't break anything for you when you try to build it. Also, the comment I left should probably be addressed at some point, just not in this PR.

**Review by mxchen2001 — APPROVED** [review]
Lgtm

**Comment by ishan0102** [comment]
Take a look at the changes to the login pytest in the most recent commit @JimothyGreene @mxchen2001

**Review by JimothyGreene — APPROVED** [review]
LGTM

**Pull request merged**